components/org.wso2.bfsi.identity.extensions/src/main/java/org/wso2/bfsi/identity/extensions/auth/extensions/response/handler/BFSICodeResponseTypeHandlerExtension.java

```diff
@@ -0,0 +1,80 @@
+/**
+ * Copyright (c) 2024, WSO2 LLC. (https://www.wso2.com).
+ * <p>
+ * WSO2 LLC. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ * <p>
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ * <p>
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.wso2.bfsi.identity.extensions.auth.extensions.response.handler;
+
+import org.wso2.bfsi.consent.management.common.util.CommonUtils;
+import org.wso2.bfsi.consent.management.common.util.Generated;
+import org.wso2.bfsi.identity.extensions.internal.IdentityExtensionsDataHolder;
+import org.wso2.carbon.identity.oauth2.IdentityOAuth2Exception;
+import org.wso2.carbon.identity.oauth2.RequestObjectException;
+import org.wso2.carbon.identity.oauth2.authz.OAuthAuthzReqMessageContext;
+import org.wso2.carbon.identity.oauth2.authz.handlers.CodeResponseTypeHandler;
+import org.wso2.carbon.identity.oauth2.dto.OAuth2AuthorizeRespDTO;
+
+/**
+ * Extension to append scope with BFSI_ prefix at the end of auth flow, before offering auth code.
+ */
+public class BFSICodeResponseTypeHandlerExtension extends CodeResponseTypeHandler {
+
+    static BFSIResponseTypeHandler bfsiResponseTypeHandler =
+            IdentityExtensionsDataHolder.getInstance().getObResponseTypeHandler();
+
+    /**
+     * Extension point to get updated scope and refresh token validity period.
+     *
+     * @param oauthAuthzMsgCtx  OAuthAuthzReqMessageContext
+     * @return OAuth2AuthorizeRespDTO
+     * @throws IdentityOAuth2Exception If an error occurred while issuing the code.
+     */
+    @Override
+    public OAuth2AuthorizeRespDTO issue(OAuthAuthzReqMessageContext oauthAuthzMsgCtx) throws IdentityOAuth2Exception {
+
+        try {
+            if (!CommonUtils.isRegulatoryApp(oauthAuthzMsgCtx.getAuthorizationReqDTO().getConsumerKey())) {
+                return issueCode(oauthAuthzMsgCtx);
+            }
+        } catch (RequestObjectException e) {
+            throw new IdentityOAuth2Exception("Error while reading regulatory property");
```

Comment:
Let's add an error log here as well if not handled at an upper level.

Reply:
Fixed in https://github.com/wso2-extensions/bfsi-consent-management/pull/6/commits/884f24f4261bc92a886a68a4e8a5bf56e7a61136